src/bindgen.rs

```diff
@@ -167,21 +147,46 @@ pub fn cargo_install_wasm_bindgen(
 
     // Remove destination directory if already exists,
     // e.g. if only the bindgen binary does not exist.
-    if destination.exists() {
-        fs::remove_dir_all(&destination).with_context(|_| {
+    if cargo_install_cache_dirname.exists() {
+        fs::remove_dir_all(&cargo_install_cache_dirname).with_context(|_| {
             format!(
                 "failed to remove existing bindgen cache directory: {}",
-                destination.display()
+                cargo_install_cache_dirname.display()
             )
         })?;
     }
 
     // Finally, move the `tmp` directory into our binary cache.
-    fs::rename(&tmp, &destination)?;
+    fs::rename(&tmp_dirname, &cargo_install_cache_dirname)?;
 
     Ok(destination_dl)
 }
 
+// Run `cargo install` to a temporary location to handle ctrl-c gracefully
+// and ensure we don't accidentally use stale files in the future
+fn cargo_install_to_tmp_dir(bin_dirname: &str, cache_root_dirname: &PathBuf, version: &str) -> Result<PathBuf, failure::Error> {
+    let tmp = cache_root_dirname.join(format!(".{}", bin_dirname));
```

Comment:
I don't think we want to use a non-`tempfile::TempDir` temporary directory. This opens the possibility for concurrency bugs (`wasm-pack build` in two different directories stepping on each others' toes). The existing method of `tempfile::TempDir` followed by an atomic rename doesn't suffer from that class of bugs.

We can use a `tempfile::TempDir` and pull this `cargo install` into its own function if the helper function takes a reference to the temp dir.

Reply:
I almost implemented this in my PR. In addition to your reasons, I am also queasy about having such a large (temporary) directory name begin with a '.'. I feel like this might not be supported on some (especially older) OSs.